### PR TITLE
feat/682: adding crypto functions to sdk package

### LINKED ISSUE #682: SDK package: Move remaining crypto functions to this package
- The vault service in extension still uses the crypto functionality directly, these can be moved into `packages/sdk`, then vault can reference the package version

### PR DESCRIPTION
**Pull request by jurevans**
Resolves #682 

- [x] Crypto functionality from `packages/crypto` added to `packages/sdk`
- [x] Connected vault service in extension to this package
- [x] Update docs
- [x] Added tests
  - [x] tests should cover example `CryptoRecord` instances as stored by the extension

### Testing

- Checkout `main`, build and install dev extension (`yarn start:chrome`), and set up some accounts
- Checkout this branch: `feat/682-crypto-sdk` and start (`yarn start:chrome`), refresh the extension in `My Extension`
- Ensure that all functions related to managing existing keys works (where encryption and decryption is concerned). And of course, installing a new extension from this branch should function as expected. The goal is to ensure that an extension store created with the previous `crypto` functionality works perfectly with the changes in this PR!

### CODE REVIEW

File: packages/sdk/src/crypto/types.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { Argon2Params as Argon2ParamsWasm, VecU8Pointer } from "@namada/crypto";
+
+export const Argon2Config = {
```
Comment (emccorson) [inline packages/sdk/src/crypto/types.ts:3]: I think `apps/extension/src/config/argon2.ts` can be deleted now.

Comment (emccorson) [reply packages/sdk/src/crypto/types.ts:3]: Also most of the types in `apps/extension/src/background/vault/types.ts`.

**Review by emccorson — APPROVED** [review]
Looks good to me!

File: packages/sdk/src/crypto/crypto.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  constructor(protected readonly cryptoMemory: WebAssembly.Memory) {}
+
+  /**
+   * @param cipherText - encrypted bytes
+   * @param params - Argon2 parameters
+   * @param iv - array of IV bytes
+   * @param salt - salt string
+   * @returns crypto record used for storage
+   */
+  private makeCryptoRecord(
```
[the remark this answers is not in the document]
Comment (mateuszjasiuk) [reply packages/sdk/src/crypto/crypto.ts:29]: I think this is a limitiation of typedoc-markdown-plugin, html docs look fine:
image

File: apps/extension/src/background/vault/tests/service.test.ts [hunk 3]
```diff
@@ -45,7 +46,12 @@ describe("Testing untouched Vault Service", () => {
     const sessionStore = new KVStoreMock<SessionPassword>(
       KVPrefix.SessionStorage
     );
-    service = new VaultService(storage, sessionStore, cryptoMemory);
+
+    const localStorage = new LocalStorage(
+      new KVStoreMock(KVPrefix.LocalStorage)
+    );
+    const sdkService = await SdkService.init(localStorage);
+    service = new VaultService(storage, sessionStore, sdkService, cryptoMemory);
```
Comment (mateuszjasiuk) [inline apps/extension/src/background/vault/tests/service.test.ts:54]: It looks like cryptoMemory is unused in VaultService so we can remove it from constructor :)

**Review by mateuszjasiuk — APPROVED** [review]
LGTM! I've noticed two more imports from `@namada/crypto`. In `keyring/messages` and `keyring/service` and we can also remove cryptoMemory from VaultService constructior :tada: :)

File: packages/sdk/src/crypto/crypto.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    return plainText;
+  }
+
+  /**
+   * Construct encryption parameters such as password hash,
+   * initialization vector, and salt from provided password
+   * @param password - required for generating password hash
+   * @returns encryption parameters
+   */
+  public makeEncryptionParams(password: string): EncryptionParams {
```
[the remark this answers is not in the document]
Comment (jurevans) [reply packages/sdk/src/crypto/crypto.ts:93]: I originally had this as public and had `encrypt` accept these params as args, but given that `encrypt` calls already this, I don't think it needs to be public - I don't see use-cases outside of what our SDK JS provides.  I ended up making this essentially the same as it was in the vault service, so I will make it private!

File: apps/extension/src/background/vault/tests/service.test.ts (hunk 3 above)
Comment (jurevans) [reply apps/extension/src/background/vault/tests/service.test.ts:54]: Thanks, got it!

**Pull request merged**